build.rs

```diff
@@ -642,6 +653,13 @@ fn cc(file: &Path, ext: &str, target: &Target, include_dir: &Path, out_file: &Pa
         }
     }
 
+    // UEFI is a baremental freestanding environment without stdlib.
+    if target.os == "uefi" && target.arch == "x86_64" {
```

Comment:
Is `target.arch == "x86_64"` a necessary condition? I guess we'd need these flags for any UEFI target, regardless of architecture.

I suggest:
```
let freestanding = target.os = UEFI;
if freestanding {
    let _ = c.flat("-ffreestanding");
}
```
Then, de-dupe this with the existing code block that adds `-nostdlibinc` and `RING_CORE_NOSTDLIBINC` by adding `freestanding || <existing conditions>`.